XenAdmin/Controls/SrPicker.cs

```diff
@@ -118,108 +118,198 @@ protected override void OnPaint(PaintEventArgs e)
             if (pool != null)
             {
                 defaultSR = _connection.Resolve(pool.default_SR);
-                pool.PropertyChanged -= Server_PropertyChanged;
-                pool.PropertyChanged += Server_PropertyChanged;
+                pool.PropertyChanged -= pool_PropertyChanged;

```

Comment:
It a safety measure to ensure the event won't be registered multiple times, because the registration happens in a handler that can be called many times while the cache changes in the background.